### ISSUE TITLE
Move to client framework (angular / vue / react)

### ISSUE THREAD

**Issue by ohadschn** [4tals/LinksForIsrael#240]
I think we may have multiple orthogonal discussion here, possibly confusing what Jekyll is, how and why we use it, what its limitations are, and how ruby plays into it.

I'm not an expert on any of these technologies (not a frontend guy), nor am I overly familiar with all our code, but I'll share my understanding. 

Jekyll is basically a glorified preprocessor that generates static websites, where static in this context means "only html/js/css & resources (typically images)"

We use it because coupled with GitHub pages, it provides free & seamless CI/CD, including excellent free hosting, which is made even better with CloudFlare (the latter is of course most optimized for static sites due to the ability to cache everything - recall there are no API calls)

Typically, Jekyll sites rely heavily on markdown, which is where this project started, but we have since done away with virtually all meaningful markdown and moved to pure HTML, CSS, and JS.

The preprocessing part is where Ruby comes in (via the liquid templating engine), and the heaviest lifting we do in this regard is inject the category/link data into the HTML and JS files.

Having said all that, it is my understanding that moving to React / Angular / Vue / etc. should be completely possible, AFAIK there is no requirement for an API backend - recall that all our resources are served statically, having been injected to our HTML/CSS from the JSON data files at build time. In other words, our "database" is git (GitHub) and we compile a snapshot of it for each site deployment. And even if we do need a database in the future, we could use something simple like firebase without changing the current Jekyll flow.

So bottom line - I suspect that almost any client framework you'd want could be used. That is of course, assuming we need it - I'm counting 4 JS files in the entire project, plus maybe a few inline JS scattered across some HTML files.

On a related note, moving to TypeScript would be an improvement for sure, and one that should be relatively painless: https://github.com/4tals/LinksForIsrael/issues/241

**Comment by 4tal** [comment]
Additional requirement is to start the new project with Typescript.

**Comment by 4tal** [comment]
Done
